### PR TITLE
#20703 Import into selected container fix

### LINKED ISSUE #20634: Wnen importing csv table mapping step not predefined after previous steps
### Description

STR

1. choose some table and from context menu select import data
2. In data transfer dialog choose CSV
3. Choose csv file from your laptop and press next btn
Actual result - on table mapping page target conteiner and source are not predifined after previous steps, also source column cut down
11

https://github.com/dbeaver/dbeaver/assets/84867187/2ecd6ea4-f7d0-4f8e-9b7e-8da43279f03e

### DBeaver Version

23.1.3

### Operating System

mac M1 monterey 12

### Database and driver

_No response_

### Steps to reproduce

_No response_

### Additional context

_No response_

### LINKED ISSUE #20703: An attempt to connect to previously chosen connection is performed during export from db to db
### Description

The application attempts to connect to the previously chosen connection before connect to chosen by user connection during export from db to db when option "Automatically reconnect to the last chosen database on export" is disabled.

### DBeaver Version

23.1.3

### Operating System

win10

### Database and driver

_No response_

### Steps to reproduce

1. Check that option "Automatically reconnect to the last chosen database on export" is disabled.
2. Perform export from db to db (target database is oracle in that case)
3. Try to perform the same export in another DB (eg PostgreSQL) - no container is chosen 
image
4. Choose target container in pg - dbeaver firstly connect to oracle (from step 2) and then connect to the target pg container
image
image

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by ShadelessFox**
Please verify against these tickets:
- https://github.com/dbeaver/dbeaver/issues/20703
- https://github.com/dbeaver/dbeaver/issues/20634

### CODE REVIEW

File: plugins/org.jkiss.dbeaver.data.transfer.ui/src/org/jkiss/dbeaver/tools/transfer/ui/pages/database/DatabaseConsumerPageMapping.java [hunk 1]
```diff
@@ -1142,10 +1142,17 @@ private DatabaseMappingContainer[] getSelectedMappingContainers() {
...
+            return;
+        }
+
         final DBPDataSourceContainer container = DatabaseConsumerSettings.getDataSourceContainer(getWizard().getSettings());
         final DBPPreferenceStore preferences = DTActivator.getDefault().getPreferences();
 
         if (preferences.getBoolean(DTConstants.PREF_RECONNECT_TO_LAST_DATABASE) || container != null && container.isConnected()) {
+            // The user is importing something into an unspecified container - reconnect to the last chosen container if the preference is set
             loadSettings(true);
         }
```
Comment (E1izabeth) [inline plugins/org.jkiss.dbeaver.data.transfer.ui/src/org/jkiss/dbeaver/tools/transfer/ui/pages/database/DatabaseConsumerPageMapping.java]: I'd simplify it, it looks confusing for me now.

```suggestion
        if (getWizard().getContainer() != null ||
            DTActivator.getDefault().getPreferences().getBoolean(DTConstants.PREF_RECONNECT_TO_LAST_DATABASE) ||
            container != null && DatabaseConsumerSettings.getDataSourceContainer(getWizard().getSettings()).isConnected()) {
            // The user is importing something into an unspecified container - reconnect to the last chosen container if the preference is set
            loadSettings(true);
        }
```

Comment (ShadelessFox) [reply plugins/org.jkiss.dbeaver.data.transfer.ui/src/org/jkiss/dbeaver/tools/transfer/ui/pages/database/DatabaseConsumerPageMapping.java]: These are two separate cases, so I separated them and added comments.

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.data.transfer.ui/src/org/jkiss/dbeaver/tools/transfer/ui/pages/database/DatabaseConsumerPageMapping.java]: You may add comments to the if branches

Comment (E1izabeth) [reply plugins/org.jkiss.dbeaver.data.transfer.ui/src/org/jkiss/dbeaver/tools/transfer/ui/pages/database/DatabaseConsumerPageMapping.java]: I'm okay if you did it intentionally, just don't like it :)

**Review by E1izabeth — APPROVED** [review]

**Review by E1izabeth — APPROVED** [review]

**Comment by MashaKorax** [comment]
verified

**Pull request merged**